### ISSUE TITLE
[BUG]: github_repository_ruleset overrides merge queue

### ISSUE THREAD

**Issue by purajit** [integrations/terraform-provider-github#2339]
### Expected Behavior

While merge queues are not yet supported, applying `github_repository_ruleset` should leave merge queue configuration untouched.

### Actual Behavior

Applying `github_repository_ruleset` disables the merge queue, basically making `github_repository_ruleset` completely unusable and useless - and even dangerous - if you are using the merge queue. The change isn't shown on the plan either (understandably). 

This is somewhat understandable since it likely edits the entire ruleset in a single query, and with merge queues not supported, this just takes default values and overrides it.

This is extraordinarily unexpected behavior. The backend should not override merge queue configurations while merge queue remains unimplemented in the API.

### Terraform Version

Terraform 1.9.2 on mac arm, linux x86, github provider v6.2.3

### Affected Resource(s)

- github_repository_ruleset

### Terraform Configuration Files

```hcl
resource "github_repository_ruleset" "repo-trunk" {
  repository  = github_repository.repo.name
  target      = "branch"
  enforcement = "active"
  ...
}
```

### Steps to Reproduce

Either import existing ruleset, or create ruleset via terraform, then enable merge queue via UI. Then,

```sh
terraform apply -target=github_repository_ruleset.repo-trunk
```

### Debug Output

```shell
N/A
```

### Panic Output

```shell
N/A
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by usmonster** [comment]
Has this been addressed by #2380? (cc: @kfcampbell @MXfive)

**Comment by MXfive** [comment]
> Has this been addressed by [#2380](https://github.com/integrations/terraform-provider-github/pull/2380)? (cc: [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/9327688?s=64&amp;u=f621def0c3137f8a1175d885b221ebbf1352ca42&amp;v=4">@kfcampbell](https://github.com/kfcampbell?rgh-link-date=2025-02-28T09%3A24%3A12.000Z) [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/3286150?s=64&amp;v=4">@MXfive](https://github.com/MXfive?rgh-link-date=2025-02-28T09%3A24%3A12.000Z))

This isn't the behaviour we experienced before using the provider I built off of that PR.  At that time it was just disregarding merge queue completely.

Merging that PR (or building yourself off of my fork) and enabling merge queue in the resource definition shuold work however.

**Comment by purajit** [comment]
Yup, this was happenign because the merge queue API had to be rolled out and then added to the terraform provider. No issues since!
